ingest/scripts/compare_hashes.py

```diff
@@ -62,60 +71,37 @@ def md5_float(string: str) -> float:
 
 
 def sample_out_hashed_records(
-    joint_insdc_accession: str,
+    joint_insdc_accession: JointInsdcAccession,
     subsample_fraction: float,
-    sampled_out: dict[str, str],
-    fasta_id: str,
-) -> tuple[float, list[dict[str, str]]]:
+    sampled_out: list[JointInsdcAccession],
+) -> tuple[float, list[JointInsdcAccession]]:
     hash_float = md5_float(joint_insdc_accession)
     keep = hash_float <= subsample_fraction
     if not keep:
-        sampled_out.append({fasta_id: joint_insdc_accession, "hash": hash_float})
+        sampled_out.append(joint_insdc_accession)
     return hash_float, sampled_out
 
 
 def process_hashes(
     ingested_insdc_accession: str,
-    fasta_id: str,
+    fasta_id: JointInsdcAccession,
     ingested_hash: str,
-    submitted: dict[InsdcAccession : dict[str, Any]],
+    submitted: dict[InsdcAccession, LatestVersion],
     update_manager: SequenceUpdateManager,
 ):
     """
     Decide if fasta_id should be submitted, revised, or noop
-    The ingested_insdc_accession corresponds to the INSDC accession base
```

Comment:
this was actually inaccurate - I have corrected the types

Reply:
My codexing identified a few clear type issues that I assume would have been clear to a linter: wondering if we'd want to add CI to check these?

Reply:
that sounds like a great idea! I didn't have the python linter enabled properly before and missed quite a few of these, so would be good to enable in a follow up PR :-)

Reply:
I created an issue for this: https://github.com/loculus-project/loculus/issues/4189